app/models/project.rb

```diff
@@ -1,6 +1,13 @@
 class Project < ApplicationRecord
-  belongs_to :user
+  belongs_to :owner, class_name: "User", foreign_key: "user_id"
   has_many :pages, -> { ordered }, dependent: :delete_all
+  has_many :collaborations, class_name: "ProjectCollaboration", dependent: :destroy
+  has_many :collaborators, through: :collaborations, source: :user
+
+  scope :editable_by, -> (user_id) {
+    left_joins(:collaborations)
+    .where("projects.user_id = :id OR project_collaborations.user_id = :id", id: user_id)
```

Comment:
```suggestion
      .where("projects.user_id = :id OR project_collaborations.user_id = :id", id: user_id)
```

Кажется, здесь не хватает DISTINCT или чего-то аналогичного. Если юзер и владеет проектом, и коллаборирует, то не будет ли он указан дважды в выдаче этого scope-а?

Reply:
Если пользователь владеет проектом, он как владелец и так может его менять и редактировать. Надо добавить валидацию, чтобы владелец проекта не мог сам себя внести в список коллабораторов - сделаю.